components/ui/table.tsx

```diff
@@ -0,0 +1,300 @@
+    const uniqueFeatureNames = new Set<string>();
+    services.forEach(service => {
+      service.features.forEach(feature => uniqueFeatureNames.add(feature.name));
+    });
+
+    // Sort feature names alphabetically for consistent row order
+    const sortedFeatureNames = Array.from(uniqueFeatureNames).sort();
+
+    return sortedFeatureNames.map(featureName => {
+      const row: TableRow = { featureName: featureName };
+      services.forEach(service => {
+        row[service.name] = service.features.find(f => f.name === featureName);
+      });
+      return row;
+    });
+  }, [services]);
+
+  // 2. define columns for TanStack Table (memoized for performance)
+  const columns = useMemo<ColumnDef<TableRow, any>[]>(() => {
+    const featureNameColumn: ColumnDef<TableRow, any> = columnHelper.accessor('featureName', {
+      id: 'featureName',
+      header: () => (
+        <div className="text-left px-4 py-2 font-semibold text-neutral-700">Feature</div>
+      ),
+      cell: info => {
+        const featureName = humanize(info.getValue());
+        return (
+          <div className="flex items-center gap-2 px-4 py-2 font-medium text-neutral-900 min-h-[80px] uppercase tracking-wider">
+            <span>{featureName}</span>
+          </div>
+        );
+      },
+      enableColumnFilter: false,
+      size: 200,
+    });
+
+    const serviceColumns: ColumnDef<TableRow, any>[] = services.map(service => {
+      const isDisabled = getColumnDisabledState(service, selectedAnswers, yamlQuestionsConfig);
+      const columnClass = isDisabled ? 'opacity-30 grayscale' : ''; 
+      
+      return columnHelper.accessor(service.name, {
+        id: service.name,
+        header: () => (
+          <div className={cn("flex flex-col items-start justify-start text-center px-2 py-2", columnClass)}>
+            <div className="font-semibold">{humanize(service.name)}</div>
+          </div>
+        ),
+        cell: info => {
+          const feature = info.getValue() as YAMLFeatureConfig | undefined;
+
+          // combine columnClass with cell-specific classes 
+          // so cell content is also affected by the column's disabled state
+          const cellContentClasses = cn("px-4 py-2 text-start", columnClass);
+
+          if (!feature) {
+            return <div className={cellContentClasses}>-</div>;
+          }
+
+          const featureClassLower = String(feature.class).toLowerCase();
+          const IconComponent = FeatureIcons[feature.name.toLowerCase()];
+          const valueColor = ClassColors[featureClassLower] || ClassColors['default'];
+
+          return (
+            <div className={cn("flex flex-col items-center justify-center p-2 min-h-[80px]", cellContentClasses)}>
+              <div className={cn("flex gap-2 font-semibold text-xl")}>
+                {/* Conditionally render IconComponent to avoid errors if it's undefined */}
+                {IconComponent && <IconComponent className={cn("text-2xl", valueColor)} />}
```

Comment:
This is a style choice really. I'm more of a fan of the following conditional render. Sometimes the `&&` could run you into trouble if the left side is a number. For example if you had something like `messageCount && <p>New messages</p>` and `messageCount` was `0` it would render `0`. 

```suggestion
                {IconComponent ? <IconComponent className={cn("text-2xl", valueColor)} /> : null}
```

See https://react.dev/learn/conditional-rendering for more!